### ISSUE TITLE
Prefs design

### ISSUE THREAD

**Issue by JasonLG1979** [pithos/pithos#345]
After playing around with rewriting our notifications and adding options to the notify plugin it seems to me that the plugins section of the prefs ui could use a little tweaking. I propose something like this:
![output](https://cloud.githubusercontent.com/assets/6667703/18611541/025a64d2-7d02-11e6-861d-672df70a8de1.gif)

1. Ditch the common plugin prefs button and put a prefs button for the plugin in the plugin's box(if it has prefs ofc)

2. Pick a unified style/size for plugin prefs dialogs.

3. do the stuff I talked about in https://github.com/pithos/pithos/issues/344

**Comment by TingPing** [comment]
Sounds good.

**Comment by JasonLG1979** [comment]
I think I'll start by cleaning up the only plugin that currently has any option, LastFM.

**Comment by JasonLG1979** [comment]
I put in a PR for changes to the lastfm ui logic. And added a template ui file that can be used by other plugins that may need prefs dialogs in the future also. There's not much to the template but it'll make sure the the plugin prefs look consistent. I'm very new to the whole template thing and I still hate Glade so be gentle, any help would be appreciated.

**Comment by TingPing** [comment]
I'm in Portland all week so It will be a while before I check it out.

**Comment by JasonLG1979** [comment]
@TingPing  I was looking at the Pandora prefs page and I had a couple thoughts.

1. The content filter setting would look better as a switch.

2. Should proxy settings be in the Pandora page? I know that it deals with how you connect with Pandora but it's not really a Pandora setting as in a setting you change in Pandora. It would clean things up.

**Comment by JasonLG1979** [comment]
I also propose that we unify the look of all the pages to match the plugins page. Here's a screenshot of the prefs in my extension with various control widgets. I think it looks rather nice. I was ofc inspired in the 1st place by the Pithos plugins page ;)

![screenshot from 2017-11-22 08-56-25](https://user-images.githubusercontent.com/6667703/33133818-9454e5f6-cf63-11e7-9c86-38454d0471c1.png)

**Comment by TingPing** [comment]
Not sure, it strays a bit from GNOME designs.

**Comment by JasonLG1979** [comment]
> Not sure, it strays a bit from GNOME designs.

Not really, My extension strays from GNOME design because it has SO MANY options but the visual design isn't jarring.

Here's a mockup of the "Pandora" minus proxy settings compared to what we have now. I would argue that the mockup is more GNOME-y than what we currently have.

![screenshot from 2017-11-24 17-44-36](https://user-images.githubusercontent.com/6667703/33225611-7074f7e6-d140-11e7-9c3f-ba2d6f22765e.png)
![screenshot from 2017-11-24 17-46-30](https://user-images.githubusercontent.com/6667703/33225613-76173934-d140-11e7-9670-02fed661e4ad.png)

**Comment by TingPing** [comment]
Our alignment is off but the current one is closer to the HIG: https://developer.gnome.org/hig/stable/visual-layout.html.en

**Comment by JasonLG1979** [comment]
Well here is a new mockup based on the HIG. It does not look as good...
![screenshot from 2017-11-25 05-15-29](https://user-images.githubusercontent.com/6667703/33230016-72c21bfe-d1a0-11e7-87a7-c3714ed340f4.png)

**Comment by JasonLG1979** [comment]
What about something like this?
![screenshot from 2017-11-25 09-44-42](https://user-images.githubusercontent.com/6667703/33232097-99b15656-d1c5-11e7-8aa9-e2df487add2b.png)

**Comment by JasonLG1979** [comment]
Another option is to *finally* add support for multiple accounts and not even show the email and password entry unless a user was adding/modifying an account.

![screenshot from 2017-11-27 11-49-45](https://user-images.githubusercontent.com/6667703/33281046-86807288-d369-11e7-8e36-d0ca5621d8d4.png)

the monkey face is ofc a stand in avatar. You can set a user profile pic in Pandora we could use that and/or the user could pick an image to be used in Pithos.

The name drop down would be a pop out like the station menu. It would allow the user to add/remove/modify accounts.

**Comment by JasonLG1979** [comment]
For bonus, it's possible to create a new account with the API so theoretically a Pithos user would never have to actually go to Pandora ever, but they would still need to create a station after they created a new account. It would work like this:

Create account > login with new account > popup station search dialog to create new station > listen to Pandora.

If I remember correctly the fields for creating a new account are:

Email
Password
age
sex
zipcode

I think the minimum age is 13? And ofc the zipcode has to be a a valid zipcode. But really it wouldn't be hard to just fill that with a random valid zipcode for anyone that doesn't have a valid zipcode or doesn't want to share theirs. It would just be a matter of keeping a list of valid zipcodes and doing a ```random.choice(zipcodes)```

**Comment by JasonLG1979** [comment]
With no Pandora link that brings us to this(this time dark):
![screenshot from 2017-11-27 12-29-12](https://user-images.githubusercontent.com/6667703/33282635-d84bd454-d36e-11e7-8e27-271467823bf2.png)

**Comment by TingPing** [comment]
Seems alright. Outside of those contents I think we need to make the dialog bigger, user a sidebar instead of tabs on top and better expose plugin options.

**Comment by JasonLG1979** [comment]
> Outside of those contents I think we need to make the dialog bigger

That's not hard.

> user a sidebar instead of tabs on top and better expose plugin options.

That's going to be tough. I much prefer compact to 10px of actual content and 250px of empty space.

It depends on if you want all the plugins on the same sidebar as everything else? We have 10 plugins plus the login and proxy settings that's 12 sidebar items so you're talking about a prefs page that would be 3 times the default size of the main window,lol!!!

**Comment by TingPing** [comment]
Yea.. I hate how smaller our main window is too :P

**Comment by JasonLG1979** [comment]
> Yea.. I hate how small our main window is too :P

It really doesn't need to be any larger. Barring a complete redesign in how Pithos works I don't see why we need to make it any larger, we present pretty much all the pertinent info. Again, I'm not in favor making the window larger for no reason. What we have currently as far as the main window is concerned is pretty serviceable. There's not much going on really, and that's a good thing.

In my mind I've always thought of Pithos as falling somewhere between an app and an applet. It's not a full fledged media player but it's not an applet. I don't mind that main window is small-ish. I think if you asked most people they'd say that that Pithos spends 99.9% of the time minimized or hidden. For me the only reason I unhide Pithos most of the time is to quit.

**Comment by JasonLG1979** [comment]
To justify a larger window we would have to present for information. Which is possible. We could emulate GNOME Music and the Pandora web interface and present stations in a cover flow, show artist bio's, provide links to purchase songs and what not.(all of which can be done currently with the API) but I'm not sure if people wouldn't see that as needless clutter/complexity?

**Comment by JasonLG1979** [comment]
The main current usability issue I see is the individual plugin prefs.

**Comment by JasonLG1979** [comment]
For the prefs options, what about a Gtk.Revealer that reveals the option if/when the plugin successfully enables?
